packages/myst-to-tex/src/proof.ts

```diff
@@ -0,0 +1,107 @@
+import { fileError, writeTexLabelledComment } from 'myst-common';
+import type { GenericNode } from 'myst-common';
+import type { Text } from 'myst-spec';
+import type { ProofContainer, ProofKind } from 'myst-ext-proof';
+import { select } from 'unist-util-select';
+import { remove } from 'unist-util-remove';
+import type { Handler } from './types.js';
+
+function kindToEnvironment(kind: ProofKind): string {
+  switch (kind) {
+    case 'theorem':
+      return 'theorem';
+    case 'proof':
+      return 'proof';
+    case 'proposition':
+      return 'proposition';
+    case 'definition':
+      return 'definition';
+    case 'example':
+      return 'example';
+    case 'remark':
+      return 'remark';
+    case 'axiom':
+      return 'axiom';
+    case 'conjecture':
+      return 'conjecture';
+    case 'lemma':
+      return 'lemma';
+    case 'observation':
+      return 'observation';
+    default:
+      return '';
+  }
+}
+
+export const proofHandler: Handler = (node, state) => {
+  state.usePackages('amsthm');
+
+  const p = node as ProofContainer;
+  const env = kindToEnvironment(p.kind);
+  if (!env) {
+    fileError(state.file, `Unhandled LaTeX proof environment "${p.kind}"`, {
+      node,
+      source: 'myst-to-tex',
+    });
+    return;
+  }
+
+  const t = select('admonitionTitle > text', node);
+  if (t) {
+    // Do not render the title twice
+    t.type = '__delete__';
+  }
+  const newNode = remove(node, '__delete__') as GenericNode;
+
+  state.write('\\begin{');
+  state.write(env);
+  state.write('}');
+  if (t) {
+    state.write('[');
+    state.write((t as Text).value);
+    state.write(']');
+  }
+  if (newNode.identifier && newNode.identifier.length > 0) {
+    state.write('\\label{');
+    state.write(newNode.identifier);
+    state.write('}');
+  }
+  state.renderChildren(newNode, true);
+  state.write('\\end{');
+  state.write(env);
+  state.write('}');
+
+  state.data.hasProofs = true;
+};
+
+export class TexProofSerializer {
+  static COMMENT_LENGTH = 50;
+
+  preamble: string;
+
+  constructor() {
+    this.preamble = this.renderThmDefinitions();
+  }
+
+  private renderThmDefinitions(): string {
+    const definitions = [
+      '\\newtheorem{theorem}{Theorem}[section]',
```

Comment:
Preamble always covers all theorem types - nice and easy. No need to worry about which ones to define for a specific document. I see how this could be made more customizable in the future, but for now, it's great 👍

Reply:
Yep, have to say that was tricky to spot, thanks to @rowanc1 for helping me here <3